src/aspire/volume/__init__.py

```diff
@@ -323,6 +449,88 @@ class FBBasisVolume(BasisVolume):
     pass
 
 
+def _gaussian_blob_Cn_vols(
+    L=8, C=2, K=16, alpha=1, order=1, seed=None, dtype=np.float64
+):
+    """
+    Generate Cn rotationally symmetric volumes composed of Gaussian blobs.
+    The volumes are symmetric about the z-axis.
+
+    Defaults to volumes with no symmetry.
+
+    :param L: The size of the volumes
+    :param C: The number of volumes to generate
+    :param K: The number of blobs each volume is composed of.
+    A Cn symmetric volume will be composed of n times K blobs.
+    :param order: The order of cyclic symmetry.
+    :param alpha: A scale factor of the blob widths
+
+    :return: A Volume instance containing C Gaussian blob volumes with Cn symmetry.
+    """
+
+    # Evaluate the symmetrized gaussian blobs.
+    def _eval_gaussians(L, K, Q, D, mu, dtype=np.float64):
+        g = grid_3d(L, dtype=dtype)
+        coords = np.array(
+            [g["x"].flatten(), g["y"].flatten(), g["z"].flatten()], dtype=dtype
+        )
+
+        n_blobs = Q.shape[0]
+        vol = np.zeros(shape=(1, coords.shape[-1])).astype(dtype)
+
+        for k in range(n_blobs):
+            coords_k = coords - mu[k, :, np.newaxis]
+            coords_k = (
+                Q[k].T @ coords_k * np.sqrt(1 / np.diag(D[k % K, :, :]))[:, np.newaxis]
+            )
+
+            vol += np.exp(-0.5 * np.sum(np.abs(coords_k) ** 2, axis=0))
+
+        vol = np.reshape(vol, g["x"].shape)
+
+        return vol
+
+    # Apply symmetry to Q and mu by generating duplicates rotated by symmetry order.
+    def _symmetrize_gaussians(Q, mu, order):
```

Comment:
Originally, I left `D` out of this function because it's not necessary to rotate/symmetrize `D`, since each symmetrized gaussian is divided by the same set of variances. However, it would probably make sense to repackage `D` in `_symmetrize_gaussians` (see my response to your comment about `% K`), so I'll make it an input and output here.

Reply:
Sounds great. Thanks!